plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -3710,57 +3710,79 @@ public void addPartition(PartitionWithStatistics partition)
             partitions.add(partition);
         }
 
-        public void execute(AcidTransaction transaction)
+        public void execute(AcidTransaction transaction, Executor executor)
         {
+            List<CompletableFuture<List<List<String>>>> futures = new ArrayList<>();
             List<List<PartitionWithStatistics>> batchedPartitions = Lists.partition(partitions, batchSize);
             for (List<PartitionWithStatistics> batch : batchedPartitions) {
+                futures.add(CompletableFuture.supplyAsync(() -> this.executeForBatch(batch), executor));
+            }
+
+            try {
+                futures.forEach(future -> createdPartitionValues.addAll(getFutureValue(future, TrinoException.class)));
+            }
+            catch (Throwable t) {
                 try {
-                    metastore.addPartitions(schemaName, tableName, batch);
```

Comment:
Why do we need to make this concurrent? It's already batched operation

Reply:
Yeah, we can increase the batch size and it will work fine for glue since it already has concurrent partition statistics updates. On the other hand hive partition statistics requests are sequential and expensive. So, we could also add concurrency there in ThriftMetastore. However, I thought that'll require more changes and extra config so maybe it's better to do it here.

Reply:
Let's add parallelism to Trift itself as in Glue